runtime/src/validate_block/implementation.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus. If not, see <http://www.gnu.org/licenses/>.
+
+//! The actual implementation of the validate block functionality.
+
+use crate::WitnessData;
+use runtime_primitives::traits::{
+	Block as BlockT, Header as HeaderT, Hash as HashT
+};
+use executive::ExecuteBlock;
+
+use substrate_trie::{MemoryDB, read_trie_value, delta_trie_root};
+
+use rstd::{slice, ptr, cmp, vec::Vec, boxed::Box, mem};
+
+use hash_db::HashDB;
+
+static mut STORAGE: Option<Box<StorageT>> = None;
```

Comment:
eep. still not a fan of the `static mut`. Why not at least a `RefCell`?

Reply:
TBH I see no reason why, with native-runtime optimization, this is guaranteed not to be used from multiple threads

Reply:
This will never be callable by the native side. First we don't support replacing the functions on native, second `register_validate_block!` resolves to nothing on native.